### PR TITLE
[Merged by Bors] - feat: Generalizing two lemmas from CommRing to CommSemiring

### PR DESCRIPTION
**Pull request by XavierXarles**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by jjaassoonn** [comment]
LGTM thanks!

**Review by jjaassoonn — APPROVED** [review]
LGTM! Thanks

File: Mathlib/AlgebraicGeometry/PrimeSpectrum/Basic.lean [hunk 1]
```diff
@@ -549,13 +549,13 @@ theorem isIrreducible_iff_vanishingIdeal_isPrime {s : Set (PrimeSpectrum R)} :
     isIrreducible_zeroLocus_iff_of_radical _ (isRadical_vanishingIdeal s)]
 #align prime_spectrum.is_irreducible_iff_vanishing_ideal_is_prime PrimeSpectrum.isIrreducible_iff_vanishingIdeal_isPrime
 
-lemma vanishingIdeal_isIrreducible {R} [CommRing R] :
+lemma vanishingIdeal_isIrreducible {R} [CommSemiring R] :
```
Comment (eric-wieser) [inline Mathlib/AlgebraicGeometry/PrimeSpectrum/Basic.lean:552]: Is `R` already a `CommSemiring` in this section?

Comment (jjaassoonn) [reply Mathlib/AlgebraicGeometry/PrimeSpectrum/Basic.lean:552]: Yes, `section CommSemiring` ends on line 707.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**